python/lsst/ts/m2gui/controltab/tab_actuator_control.py

```diff
@@ -33,7 +57,516 @@ class TabActuatorControl(TabDefault):
+            ),
+        }
+
+        self._set_widget_and_layout()
+
+        self._set_signal_script(self.model.signal_script)
+        self._set_signal_detailed_force(
+            self.model.utility_monitor.signal_detailed_force
+        )
+
+    def _create_progress_bar(self, value_min, value_max, value_init):
+        """Create the progress bar. This will show the execution progress of
+        script in the cell controller.
+
+        Parameters
+        ----------
+        value_min : `float` or `int`
+            Minimum value.
+        value_max : `float` or `int`
+            Maximum value.
+        value_init : `float` or `int`
+            Initial value.
+
+        Returns
+        -------
+        progress_bar : `PySide2.QtWidgets.QProgressBar`
+            Progress bar.
+        """
+
+        progress_bar = QProgressBar()
+        progress_bar.setRange(value_min, value_max)
+        progress_bar.setValue(value_init)
+
+        return progress_bar
+
+    @Slot()
+    def _callback_script_load_script(self, file_name=""):
+        """Callback of the load-script button in script control. The cell
+        controller will load the script.
+
+        Parameters
+        ----------
+        file_name : `str`, optional
+            File name. This is used for the unit test only. (the default is "")
+
+        Returns
+        -------
+        name : `str`
+            Script name.
+        """
+
+        if file_name == "":
+            file_name = QFileDialog.getOpenFileName(caption="Select script file")[0]
+
+        name = ""
+        if file_name != "":
+            # Only the file name is important
+            file_path = Path(file_name)
+            name = file_path.name
+
+            is_successful = run_command(
+                self.model.command_script, CommandScript.LoadScript, name
+            )
+
+            if is_successful:
+                self._info_script["file"].setText(file_name)
+
+        return name
+
+    @Slot()
+    def _callback_script_command(self, command):
+        """Callback of the command buttons in script control. The cell
+        controller will run the related command.
+
+        Parameters
+        ----------
+        command : enum `CommandScript`
+            Script command.
+        """
+
+        run_command(self.model.command_script, command)
+
+        if command == CommandScript.Clear:
+            self._info_script["file"].setText("")
+            self._info_script["progress"].setValue(0)
+
+    def _create_target_displacement(self):
+        """Create the target displacement of actuator.
+
+        Returns
+        -------
+        displacement : `QDoubleSpinBox`
+            Target displacement of the actuator.
+        """
+
+        displacement = QDoubleSpinBox()
+        displacement.setDecimals(
+            self.model.utility_monitor.NUM_DIGIT_AFTER_DECIMAL_DISPLACEMENT
+        )
+        displacement.setRange(-self.MAX_DISPLACEMENT, self.MAX_DISPLACEMENT)
```

Comment:
The transformation between the step and millimeter  will be done in the M2 cell side instead of GUI, which tris to hold the minimum information of the calculation. As an educated guess, I decide to set the maximum allowed movement to be 20 cm. Unfortunately, we do not have the details evaluation of allowed displacement. There is the plan to evaluate this in the system engineer team but it will not be soon.

Reply:
Updated in the commit: [854d782](https://github.com/lsst-ts/ts_m2gui/pull/12/commits/854d7823b36c3edc5c9f8408fe5d0011a77139ea). Thanks!